Features/bootstrap/MainContext.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+use QATools\BehatExtension\Context\QAToolsContext;
+
+class MainContext extends QAToolsContext
+{
+
+	/**
+	 * @When /^the user enters (.+?) and (.+?)$/
+	 */
+	public function fillForm($username, $password)
+	{
+		$this->page->fillForm($username, $password);
+	}
+
+	/**
+	 * @Then /^the login should fail$/
+	 */
+	public function loginFail()
+	{
+		$this->page = $this->page->loginFailure();
```

Comment:
The method on PageObject/element class represents complete action from UI/user viewpoint. User sees login process as single action: fill fields + press ENTER/login button. This way the PageObject should do it in same fashion.